projects/step-frontend/src/lib/modules/timeseries/util/timeseries-colors-pool.ts

```diff
@@ -17,6 +17,8 @@ export class TimeseriesColorsPool {
    * @param key - If it is not new, the existing value will be returned, otherwise a new color is generated.
    */
   assignColor(key: string): string {
```

Comment:
There is redundant code:

`
assignColor(key: string): string {
    let color = this.randomRGBA();
    if (this.predefinedColors.length !== 0) {
      color = this.predefinedColors[0];
      this.predefinedColors.shift();
    }
    this.assignedColors[key] = color;
    return color;
  }
`

Reply:
@neogucky I don't think this is redundant. The randomRGBA() would be called all the time, even though 90% of the time it goes on the second condition. I don't like to have functions called if they are not needed, even if the complexity of the function is not very big...